### ISSUE TITLE
Refactor/merge Visual and Panels mode

### ISSUE THREAD

**Issue by XVilka** [rizinorg/rizin#384]
Currently, both modes are implemented completely independently, sometimes with code duplication/copy-paste.
I propose to reduce the code and feature duplication by:
1. Migrate **Visual** mode from commands to the API: https://github.com/rizinorg/rizin/issues/382
2. Migrate **Panels** mode from command to the API: https://github.com/rizinorg/rizin/issues/383
3. Migrate **Visual** menus code from commands to the API: https://github.com/rizinorg/rizin/issues/491
4. Only after the three above are finished, determine what code is logically/semantically share by both and extract it into other file or RzCons or RzUtils.
5. Determine what features overlap, and consider the merge of both modes.

Please note, that many features that are present in **Visual** mode are missing in the **Panels** mode, and there is a significant difference how these modes operate:
- Visual mode has the concept of *rotation* - press `p`/`P` to switch between representations
- Panels mode has the concept of *panes* and useful for custom arrangement.

Thus, we might consider keeping the both modes as is from the user point of view, but reduce the code and feature duplication to the minimum.

**Comment by ret2libc** [comment]
I think this is not going to happen anytime "soon" and by soon I mean at least a year, considering the many projects we have opened right now. I propose to close this issue for now as a way to try to keep the list of issues manageable.

**Comment by XVilka** [comment]
No, it's important issue and there is a lot of duplicated code.
